File: programmers/level1/033.py
# 33. 3진법 뒤집기
def solution(n):
    answer = 0
    lst = []
    i = 16
    start = False
    while n>0:
        v=n//(3**i)
        if v!=0:
            start=True
        if start:
            lst.append(v)
            n-=v*(3**i)
        i-=1
        
    for i in range(len(lst)):
        answer+=lst[i]*(3**i)
    return answer

File: programmers/level2/033.java
// 33. 피로도
import java.util.*;
class Solution {
    
    public int solution(int k, int[][] dungeons) {
        final int N = dungeons.length;
        boolean[] visited = new boolean[N];    
        int[][] output= new int[N][2];
        int answer = -1;
        
        for(int i=0 ;i<factorial(N); i++){
            Collections.shuffle(Arrays.asList(dungeons));
            answer= Math.max(answer, search(0,k,dungeons));
        }
        
        return answer;
    }
    int factorial(int n){
        int rslt =1;
        if(n==0)
            return rslt;
        for(int i=2; i<=n;i++){
            rslt*=i;
        }
        return rslt;
    }
    
    int search(int idx, int k,int[][] dungeons){
        if(idx== dungeons.length)
            return 0;
        if(k>= dungeons[idx][0]){
            return Math.max(search(idx+1, k-dungeons[idx][1], dungeons)+1, search(idx+1, k, dungeons));
        }else{
            return search(idx+1, k, dungeons);
        }
        
    }
}